### PR TITLE
AMK-49 store headers in task

### PR DESCRIPTION
**Pull request by rashi-j**
## Change description

> Added headers to task document
> Added keyword field within __task_status to make it queryable.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java:401]: Do not change `isStringField` for `TASK_STATUS`, keep it false to be backward compatible with existing tenants, it should be `default:text` & `keyword` as field

```
"__task_status": {
    "type": "text",
    "fields": {
        "keyword": {
            "type": "keyword"
        }
    },
    "copy_to": [
        "all"
    ]
}

File: common/src/main/java/org/apache/atlas/repository/Constants.java
Comment (nikhilbonte21) [inline common/src/main/java/org/apache/atlas/repository/Constants.java:377]: TASK_HEADER_ATLAN_VIA_UI is missing in this map

File: common/src/main/java/org/apache/atlas/repository/Constants.java
Comment (nikhilbonte21) [inline common/src/main/java/org/apache/atlas/repository/Constants.java:370]: This can be simple Set instead of Map as the check I saw only validating as key present in map

```
if (Constants.TASK_HEADER_MAP.containsKey(key)) {
    ...
}

**Review by nikhilbonte21 — CHANGES_REQUESTED** [review]

**Review by hr2904 — APPROVED** [review]
LGTM

**Review by nikhilbonte21 — APPROVED** [review]
LGTM, Wait for @sumandas0 approval

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java:401]: Just for reminder this change will only work for new tenants only! Not for old tenants as the mapping in ES already been created.

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
Comment (sumandas0) [inline repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java:336]: Could have just used KEYWORD_MULTIFIELD? Intead of creating new one? NOT A BLOCKER!

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java:401]: SO you need to run simple migration to add this field in ES mapping in all the existing tenants, we already did this. NOTE: This is just mapping update won't take much time.
Then if we want this for other existing vertex we need reindex the fields.

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
Comment (rashi-j) [reply repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java:336]: KEYWORD_MULTIFIELD also adds an analyser by itself, hence we decided to create one without that. Analysing/supporting different cases won't be needed here, and it could anyway be an overhead.

File: repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
Comment (rashi-j) [reply repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java:401]: That serves the purpose, atleast in playbook use-case we have nothing to do with past tasks that are already done. This [PR](https://github.com/atlanhq/marketplace-packages/pull/12604/files) should handle adding the mapping correct?

**Comment by rashi-j** [comment]
Attaching proof of API working as expected for task search via headers and status:

**Comment by rashi-j** [comment]
Screenshot 2024-11-14 at 10 54 47 AM

**Review by sumandas0 — APPROVED** [review]

**Pull request merged**
